src/loki/search.cc

```diff
@@ -659,20 +631,36 @@ struct bin_handler_t {
           const GraphTile* other_tile;
           auto opposing_edge = reader.GetOpposingEdge(candidate.edge_id, other_tile);
           if(!other_tile) continue; //TODO: do an edge snap instead, but you'll only get one direction
-          correlate_node(pp.location, opposing_edge->endnode(), candidate, correlated);
+          correlate_node(pp.location, opposing_edge->endnode(), candidate, correlated, filtered);
         }//it was the end node
         else if((back && candidate.edge->forward()) || (front && !candidate.edge->forward())) {
-          correlate_node(pp.location, candidate.edge->endnode(), candidate, correlated);
+          correlate_node(pp.location, candidate.edge->endnode(), candidate, correlated, filtered);
         }//it was along the edge
         else {
-          correlate_edge(pp.location, candidate, correlated);
+          correlate_edge(pp.location, candidate, correlated, filtered);
         }
       }
+
+      //if we have nothing because of heading we'll just ignore it
```

Comment:
instead of having the fallback happening at each call, we just do the fallback at the end of correlating all the locations